### PR TITLE
Consolidate common R2R vs. non-R2R code

### PR DESCRIPTION
**Pull request by uartie**
The first encode/decode command is always the same for first call in r2r and non-r2r cases.  Thus, share that code.  Also, cleanup other unnecessary complexities.

### CODE REVIEW

**Comment by uartie** [comment]
@haribommi can you please test if this all works on your side?

File: test/ffmpeg-qsv/encode/encoder.py [hunk 1]
```diff
@@ -146,15 +140,14 @@ def check_output(self):
...
-    name = self.gen_name().format(**vars(self))
-    self.decoded = get_media()._test_artifact(
-      "{}-{width}x{height}-{format}.yuv".format(name, **vars(self)))
-
-    call(
-      "ffmpeg -hwaccel qsv -hwaccel_device /dev/dri/renderD128 -v verbose"
-      " -c:v {ffdecoder} -i {encoded} -vf 'hwdownload,format={hwformat}'"
-      " -pix_fmt {mformat} -f rawvideo -vsync passthrough -vframes {frames}"
-      " -y {decoded}".format(**vars(self)))
+    oopts = "-c:v {ffdecoder} -i {encoded}"
```
Comment (haribommi) [inline test/ffmpeg-qsv/encode/encoder.py]: assignment error, oopts should be changed to iopts and vice-versa

**Comment by haribommi** [comment]
> @haribommi can you please test if this all works on your side?

except ffmpeg-qsv encode, all other changes are working fine no test run errors

File: test/ffmpeg-qsv/encode/encoder.py (hunk 1 above)
Comment (uartie) [reply test/ffmpeg-qsv/encode/encoder.py]: Good catch.  I guess I only tested with jpeg encode (which I forgot overrides this method).

Fixed.

**Comment by haribommi** [comment]
changes LGTM

**Comment by uartie** [comment]
@haribommi thanks for testing and reviewing

**Pull request merged**